api/src/errors/invalid-config.ts

```diff
@@ -0,0 +1,13 @@
+import { createError } from '@directus/errors';
+import { ErrorCode } from './codes.js';
+
+export interface InvalidConfigErrorExtensions {
+	provider: string;
+	reason?: string;
+}
+
+export const InvalidConfigError = createError<InvalidConfigErrorExtensions>(
```

Comment:
Should we rename this Error to something like `InvalidProviderError` as it seems very specific to Provider and not Config?